src/core/raster/qgsrasterfilewriter.cpp

```diff
@@ -231,9 +231,7 @@ QgsRasterFileWriter::WriterError QgsRasterFileWriter::writeDataRaster( const Qgs
       QgsRasterProjector *projector = pipe->projector();
       if ( projector && projector->destinationCrs() != projector->sourceCrs() )
       {
-        Q_NOWARN_DEPRECATED_PUSH
-        QgsCoordinateTransform ct( projector->destinationCrs(), projector->sourceCrs() );
-        Q_NOWARN_DEPRECATED_POP
+        QgsCoordinateTransform ct( projector->destinationCrs(), projector->sourceCrs(), srcProvider->transformContext() );
```

Comment:
I think it'd be cleaner if this was set in QgsRasterFileWriter::writeRaster

Reply:
@nyalldawson ok, this class qgsrasterfilewriter.cpp is like the Pandora's box. 
There are raster provider factories that do not accept raster options and do not set the transform context. 
What do you suggest to do here? Should it be responsibility of the callers to set the context in the provider?

Reply:
I might be misunderstanding the issue -- but I don't think these factories will need the context. It's only the pipe with the optional transformer which would need them. (Otherwise can you point to the troublesome code?)

Reply:
@nyalldawson the potential issue here is that, in general, the new approach is to construct a raster layer by passing LayerOptions that contains the tc and store it into the provider (because we need it for cloning), the factories are constructing a data provider that do not set the tc, so the caller will be responsible for setting it when the data provider is used to construct a layer. We can leave it as it is but I thought it was important to discuss it.

Additionally passing the tc in `writeRaster` is not enough, because it uses `writeImageRaster` that requires a context, but this is not an issue.

See if the last commit works.

Reply:
Looks good to me, are you happy?

Reply:
Happy like Easter ... (Italian way of saying that means very happy)